sycl/doc/extensions/ExplicitSIMD/dpcpp-explicit-simd.md

```diff
@@ -0,0 +1,776 @@
+  simd<uint16_t, N> operator >=(const simd &RHS) const;
+  simd<uint16_t, N> operator <(const simd &RHS) const;
+  simd<uint16_t, N> operator <=(const simd &RHS) const;
+  simd<uint16_t, N> operator ==(const simd &RHS) const;
+  simd<uint16_t, N> operator !=(const simd &RHS) const;
+
+  // Select operations.
+  template <int Size, int Stride>
+  simd<Ty, Size> select(uint16_t Offset = 0) &&;
+  template <int Size, int Stride>
+  simd_view<simd, region1d_t<Ty, Size, Stride>>
+  select(uint16_t i = 0) &;
+
+  // Replicate operations.
+  template <int Rep> simd<Ty, Rep * N> replicate();
+  template <int Rep, int W> simd<Ty, Rep * W> replicate(uint16_t i);
+  template <int Rep, int VS, int W>
+  simd<Ty, Rep * W> replicate(uint16_t i);
+  template <int Rep, int VS, int W, int HS>
+  simd<Ty, Rep * W> replicate(uint16_t i);
+
+  // Format operations.
+  template <typename EltTy> auto format() &;
+  template <typename EltTy, int Height, int Width> auto format() &;
+
+  // Merge operations.
+  void merge(const value_type &Val, const mask_type_t<N> &Mask);
+  void merge(const value_type &Val1, value_type Val2,
+             const mask_type_t<N> &Mask);
+} // namespace gpu
+} // namespace intel
+} // namespace sycl
+```
+
+Every specialization of ```sycl::intel::gpu::simd``` shall be a complete type. The term
+simd type refers to all supported specialization of the simd class template.
+To access the i-th individual data element in a simd vector, Explicit SIMD supports the
+standard subscript operator ```[]```, which returns by value.
+
+For simd type object, Explicit SIMD supports the following simd vector operations:
+- Unary operators: ++ (*pre-/post-increment*), -- (*pre-/post-decrement*)
+- Binary operators: +, -, *, /, &, |, ^
+- Compound assignments: +=, -=, *=, /=, &=, |=, ^=
+- Compare operators: >, >=, <, <=, ==, !=
+
+These are all element-wise operations, which apply a specified operation to the
+elements of one or more simd objects and should follow the standard C++ rules for
+the corresponding scalar data element computation. Each such application is
+unsequenced with respect to the others.
+
+To reference a subset of the elements in simd vector object, Explicit SIMD provides ```select```
+function, which returns a `simd` or `simd_view` object (*described below*) representing
+the selected sub-vector starting from the i-th element. The number of selected
+elements is specified by the template parameter **Size**, while the distance
+between two adjacent elements is specified by the template parameter **Stride**.
+
+```cpp
+  simd<int, 8> a;
+  simd<int, 4> b;
+  // ...
+  b = a.select<4, 2>(1);  // size=4, stride=2, offset=1 (elements a(1),
+                          // a(3), a(5) and a(7) are copied to b)
+```
+
+<p align="center">
+<img src="images/VectorOdd.svg" title="1D select example" width="600" height="60"/>
+</p>
+
+```cpp
+a.select<4, 2>(0) = b;  // selected elements of a are replaced
```

Comment:
The naming sounds strange for something that can be in LHS. On the other hand I do not have another good proposition...